sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -7,227 +7,182 @@
 
 from sklearn.decomposition import TruncatedSVD, PCA
 from sklearn.utils import check_random_state
-from sklearn.utils.testing import (assert_array_almost_equal, assert_equal,
-                                   assert_raises, assert_greater,
-                                   assert_array_less, assert_allclose)
+from sklearn.utils.testing import assert_array_less, assert_allclose
 
+SVD_SOLVERS = ['arpack', 'randomized']
 
-# Make an X that looks somewhat like a small tf-idf matrix.
-# XXX newer versions of SciPy >0.16 have scipy.sparse.rand for this.
-shape = 60, 55
-n_samples, n_features = shape
-rng = check_random_state(42)
-X = rng.randint(-100, 20, np.product(shape)).reshape(shape)
-X = sp.csr_matrix(np.maximum(X, 0), dtype=np.float64)
-X.data[:] = 1 + np.log(X.data)
-Xdense = X.A
 
+@pytest.fixture(scope='module')
+def X_sparse():
```

Comment:
I don't see why not. 

The only risk is in place modification of the data. Common tests that check read-only arrays as input should prevent it somewhat. Still one way around it could be a module or session scoped fixture that load and prepares the data, and a function scoped fixture that makes a copy of it. Another alternative could be to return data array with a read-only=True flag.

Still that's something we could look into separately. Here that should be fairly equivalent to using a global variable with data.